main/src/ca/uwaterloo/flix/language/phase/TypeInference.scala

```diff
@@ -889,7 +889,10 @@ object TypeInference {
       case KindedAst.Expr.TryCatch(exp, rules, loc) =>
         val rulesType = rules map {
           case KindedAst.CatchRule(sym, clazz, body) =>
-            visitExp(body)
+            for {
+              _ <- unifyTypeM(sym.tvar, Type.mkNative(clazz, sym.loc), sym.loc)
```

Comment:
What about being an instance of Throwable?

Reply:
Does this also need to be fixed in ConstraintGeneration?

Reply:
> What about being an instance of Throwable?

I don't know how to do that. This PR still improves things though

Reply:
CC @mlutze

Reply:
I guess your test will help Matt make the required refactor.

Reply:
Ive corrected the constraint generator